ethcore/verification/src/queue/mod.rs

```diff
@@ -500,6 +499,8 @@ impl<K: Kind, C> VerificationQueue<K, C> {
 				Ok(hash)
 			},
 			Err((input, err)) => {
+				// NOTE: we could drop `processing RwLock` here but it's not dropped to ensure
+				// that the `verification.bad` is written to before the next input is processed
```

Comment:
Aren't we already holding the lock to `self.verification.bad` here, from L479? Wouldn't taking it again on L515 or L518 lead to a deadlock?

Reply:
Great catch, let me put in a block so it is dropped and reclaimed here to use the same behavior as before.